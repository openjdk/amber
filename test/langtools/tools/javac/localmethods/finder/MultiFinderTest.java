/**
 * @test /nodynamiccopyright/
 * @bug 8233754
 * @compile/ref=MultiFinderTest.out -XDrawDiagnostics -XDfind=all MultiFinderTest.java
 */

public class MultiFinderTest<X> {

    MultiFinderTest<String> mf = new MultiFinderTest<String>();

    interface SAM {
        void m();
    }

    SAM s = new SAM() { public void m() { } };

    static <Z> Z id(Z z) { return z; }

    private static void test(String a) {
        String s = MultiFinderTest.<String>id(a);
    }

    public static void main(String [] args) {
        for (String s : args) {
            test(s);
        }
    }
}
