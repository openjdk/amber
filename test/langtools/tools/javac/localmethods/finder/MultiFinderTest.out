MultiFinderTest.java:19:25: compiler.warn.method.could.be.declared.local: main(java.lang.String[]), kindname.class, MultiFinderTest
MultiFinderTest.java:9:53: compiler.warn.diamond.redundant.args
MultiFinderTest.java:15:23: compiler.warn.potential.lambda.found
MultiFinderTest.java:20:16: compiler.warn.local.redundant.type
MultiFinderTest.java:20:46: compiler.warn.method.redundant.typeargs
MultiFinderTest.java:24:21: compiler.warn.local.redundant.type
6 warnings
